jaeger-core/src/test/java/io/jaegertracing/senders/InMemorySender.java

```diff
@@ -34,33 +32,32 @@
   private Semaphore semaphore = new Semaphore(Integer.MAX_VALUE);
 
   public InMemorySender() {
-    appended = new ArrayList<Span>();
-    flushed = new ArrayList<Span>();
-    received = new ArrayList<Span>();
+    appended = new ArrayList<>();
+    flushed = new ArrayList<>();
+    received = new ArrayList<>();
   }
 
   public List<Span> getAppended() {
-    return new ArrayList<Span>(appended);
+    return new ArrayList<>(appended);
```

Comment:
what are the threading expectations?

Reply:
Given that it's used only for tests, I would say that we don't guarantee anything. Concurrent modification *will* cause a `ConcurrentModificationException`, for instance.